### PR TITLE
Adding responsive wizard guidelines

### PR DESCRIPTION
**Pull request by cshinn**
## Description
- Adds a section describing the mobile presentation of a wizard
- Closes issue #347 
- **Will be updated with high fidelity images when those become available**
## Changes
- Adds a Responsive States section with mobile guidelines
- Adds an image of the mobile wizard to the overview page

### CODE REVIEW

**Comment by serenamarie125** [comment]
@cshinn can you add a couple things here?

1 - an image without subtitles (similar to below)
responsivewizardtitle

2 - a note indicating that the caret and display of where you are / and allowing navigation is optional behavior?

I think that we talked about having some testing down on whether this would be an overlay or actually push content down.  Unsure if @mcarrano @Mgranfie are ok with us adding this specific piece now, or waiting til after testing is complete?  @LHinson

**Comment by cshinn** [comment]
@serenamarie125 I've added a new section with a simplified mobile wizard

**Comment by Mgranfie** [comment]
@cshinn @serenamarie125 @mcarrano This is great Chris and does a great job covering the use cases and identifying what is not covered, at this time.

1. In response to your question about the caret drop down, I expected it to occupy the entire page. That would bring focus and consistency, as well as, a permanent extensibility solution to the user. This is pretty common on mobile too...
2. Question on content of knowing how many steps exist, when in the simplified view. Not sure if this is necessary, just wanted to see if anyone else had any use cases or thoughts to drive a direction here?
3. Question on the breadcrumbs and an overflow solution, and if any thought was given to that already?

**Comment by serenamarie125** [comment]
@mcarrano 
re: your first point, yes this is the responsive state.  I will added an inline comment to make that suggestion in the design documentation
re: 3rd point (expanding as an overlay) I thought our discussions the other day talked about doing some prototyping and testing before actual implementation.

**Comment by serenamarie125** [comment]
@Mgranfie 
re: #2 we were going to try and get some user feedback through our OpenShift DXP meetings regarding is it good enough to say 
"2: Step Name" or should we be saying "2 of 8: Step Name"

re: #3 hmmm i'm not sure what you mean by breadcrumbs as an overflow solution - could you provide an example?   What would be shown in the breadcrumb?

**Comment by cshinn** [comment]
@Mgranfie I've updated the docs with some notes on truncation and on step naming in general

**Comment by LHinson** [comment]
@cshinn It would be great to see that documentation reflect the implementation decision to require the user to scroll the page to get to the buttons if content is taller than the screen.

**Comment by serenamarie125** [comment]
I agree with this approach.  Thank you

**Comment by cshinn** [comment]
@LHinson I've updated the guidelines to include the information in the blog post

File: pattern-library/communication/wizard/design/design.md [hunk 1]
```diff
@@ -56,13 +58,35 @@ Modal wizards should be used when the task is non-necessary or is one of several
...
-The following functionality is not covered in this pattern:
+## Responsive States
+
+### Mobile Wizard
+![Collapsed responsive wizard](img/responsive-wizard-callout1.png)
+
+1. **Current Main Step:** The currently active main step is displayed at the top of the form along with the step number
+1. **Current Sub-step** (Optional)**:** If the current main step has sub-steps, the name of the sub-step appears next to it at the top of the page. If the main and sub-step names are long enough that [truncation](https://www.patternfly.org/styles/terminology-and-wording/#truncation) is required, the sub-step name should be truncated before the main step name unless the sub-step name is critical to filling in the form.
+1. **Steps Dropdown:** Clicking on this dropdown reveals all steps in the wizard and enables users to switch between them if applicable
+1. **Button Bar:** Wizard actions are available on the button bar, which is fixed at the bottom of the page for embedded wizards and follows the guidelines outlined in [this blog post](https://blog.patternfly.org/how-we-scroll/) for modal scrolling otherwise.
```
Comment (LHinson) [inline pattern-library/communication/wizard/design/design.md]: @cshinn This sentence contradicts itself with the former specifying to used fixed button bar and the former to point to the blog which specifies to scroll. If you want the documentation to match the implementation then it should read something like: 

>Wizard actions are available on the button bar, which is fixed at the bottom of the page for larger screens but will scroll on mobile.

FYI @serenamarie125 @matthewcarleton

Comment (matthewcarleton) [reply pattern-library/communication/wizard/design/design.md]: Yes that sounds more accurate.

Comment (cshinn) [reply pattern-library/communication/wizard/design/design.md]: @LHinson My intention was that for wizards that are embedded in the page, buttons would be fixed, but in a modal, they would follow our current modal scrolling guidelines. I can reword this if that's not clear

**Comment by LHinson** [comment]
@cshinn The current implementation of the wizard has fixed buttons in the desktop version however requires scroll in the responsive state. I think the document currently articulates that the wizard modal scroll no matter which state. 

>Wizard actions are available on the button bar, which is fixed at the bottom of the page for embedded wizards and modals wizards except for in smaller screen sizes, which may require the user to scroll the page for the button bar.

**Comment by cshinn** [comment]
@LHinson I can't seem to find that quote anywhere in the current documentation. Which section did it come from?

**Comment by LHinson** [comment]
Apologies @cshinn. It currently says: 

> Wizard actions are available on the button bar, which is fixed at the bottom of the page for embedded wizards and follows the guidelines outlined in [this blog post](https://blog.patternfly.org/how-we-scroll/) for scrolling in modal wizards.

This is incorrect because the modal implementation has a fixed button bar except for in the smaller screens so I was suggesting for it to say:

> Wizard actions are available on the button bar, which is fixed at the bottom of the page for embedded wizards and modals wizards except for in smaller screen sizes, which requires the user to scroll the page for the button bar.

**Comment by serenamarie125** [comment]
@LHinson, it looks like @cshinn made an update to address your request.  Are we ready to merge, or do we need someone else to review?

**Comment by LHinson** [comment]
@cshinn thanks for updating the text!

@serenamarie125 I think we can merge. agree @mcarrano ? 

I have filed an issue regarding the discrepancy between the images and the behavior. https://github.com/patternfly/patternfly-design/issues/422 I think this updated can be made separate from this PR.

**Comment by mcarrano** [comment]
LGTM! I will merge.

**Pull request merged**